src/main/java/me/ryanhamshire/GriefPrevention/UUIDFetcher.java

```diff
@@ -98,6 +99,18 @@ public void call() throws Exception
         //for online mode, call Mojang to resolve the rest
         if (GriefPrevention.instance.getServer().getOnlineMode())
         {
+            Pattern validNamePattern = Pattern.compile("^\\w+$");
+
+            // Don't bother requesting UUIDs for invalid names from Mojang.
+            names.removeIf(name ->
+            {
+                if (name != null && name.length() >= 3 && name.length() <= 16 && validNamePattern.matcher(name).find())
```

Comment:
Isn't it easier to include the length checks in the regex pattern?

[a-zA-Z0-9_]{3,16}

Reply:
Easier, yes. Also much slower.

Reply:
Wouldn't it be better to just compile this in a class variable/static context? Compiling regex is expensive in function calls. If the regex is pre-compiled, performance-wise the length checks won't matter.

Reply:
I figure one compile per call on a method that internally only gets called once is technically better. If it were something that's expected to be in frequent use, I would definitely agree.